scripts/release-notes.js

```diff
@@ -0,0 +1,127 @@
+const {
+  exec,
+  releaseYargsBuilder,
+  requireGitHubCLI,
+  buildQuery,
+} = require('./release-utils');
+const diffFeatures = require('./diff-features');
+
+function main(argv) {
+  const { startVersionTag, endVersionTag } = argv;
+
+  requireGitHubCLI();
+
+  const allAdds = [];
+  const allRemoves = [];
+
+  console.error(
+    `Generating release notes from ${startVersionTag} to ${endVersionTag}`,
+  );
+  for (const pull of pullsFromGitHub(startVersionTag, endVersionTag)) {
+    process.stderr.write(`Diffing features for #${pull.number}`);
+
+    const diff = diffFeatures({ ref1: pull.mergeCommit });
+
+    console.error(
+      ` (${diff.added.length} added, ${diff.removed.length} removed)`,
+    );
+
+    for (const feature of diff.added) {
+      allAdds.push({
+        number: pull.number,
+        url: pull.url,
+        feature,
+      });
+    }
+
+    for (const feature of diff.removed) {
+      allRemoves.push({
+        number: pull.number,
+        url: pull.url,
+        feature,
+      });
+    }
+  }
+
+  console.error();
+
+  allRemoves.sort((a, b) => a.feature.localeCompare(b.feature));
+  allAdds.sort((a, b) => a.feature.localeCompare(b.feature));
+
+  console.log(preamble());
+  console.log(markdownifyChanges(allRemoves, allAdds));
+  console.log('<!-- TODO: replace with `npm run release-stats` -->');
+}
+
+function pullsFromGitHub(start, end) {
+  const searchDetails = {
+    limit: 1000, // As many PRs as GitHub will allow
+    search: `${buildQuery(end, start, false)}`,
+    json: `number,url,mergeCommit`,
+    jq: '[.[] | { mergeCommit: .mergeCommit.oid, number: .number, url: .url }]', // Flatten the structure provided by GitHub
+  };
+  const args = Object.entries(searchDetails)
+    .map(([key, value]) => `--${key}='${value}'`)
+    .join(' ');
+  const command = `gh pr list ${args}`;
```

Comment:
So we shell out to GitHub here. Perhaps this ought to use the GitHub API package, but shelling out has some advantages: we don't have to do pagination ourselves and we don't have to deal with auth. Plus, I had already written a shell script to do a lot of this stuff and this was easy to get up and running quickly.